src/encounter.c

```diff
@@ -54,982 +55,951 @@
-    v2->trainerIDs[2] = param3;
+    dto->trainerIDs[BATTLER_ENEMY_SLOT_1] = enemyTrainer1ID;
+    dto->trainerIDs[BATTLER_ENEMY_SLOT_2] = enemyTrainer2ID;
+    dto->trainerIDs[BATTLER_PLAYER_SLOT_2] = partnerTrainerID;
 
-    TrainerData_Encounter(v2, fieldSystem->saveData, param4);
-    GameRecords_IncrementRecordValue(SaveData_GetGameRecordsPtr(fieldSystem->saveData), RECORD_UNK_008);
-    sub_02050C4C(taskMan, v2, EncEffects_CutInEffect(v2), EncEffects_BGM(v2), param5);
+    TrainerData_Encounter(dto, fieldSystem->saveData, heapID);
+    GameRecords_IncrementRecordValue(SaveData_GetGameRecordsPtr(fieldSystem->saveData), RECORD_TRAINER_BATTLES_FOUGHT);
+    StartEncounter(taskMan, dto, EncEffects_CutInEffect(dto), EncEffects_BGM(dto), resultMaskPtr);
 }
 
-void sub_0205167C(FieldTask *taskMan, const u8 *param1, int param2)
+void Encounter_NewVsLink(FieldTask *task, const u8 *partyOrder, int battleType)
 {
-    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(taskMan);
-    UnkStruct_02050ACC *v1;
-    FieldBattleDTO *v2;
-
-    v2 = FieldBattleDTO_New(11, param2);
-    FieldBattleDTO_InitWithPartyOrderFromSave(v2, fieldSystem, param1);
+    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(task);
+    Encounter *encounter;
+    FieldBattleDTO *dto = FieldBattleDTO_New(HEAP_ID_FIELDMAP, battleType);
+    FieldBattleDTO_InitWithPartyOrderFromSave(dto, fieldSystem, partyOrder);
 
-    v1 = sub_02050ACC(v2, EncEffects_CutInEffect(v2), EncEffects_BGM(v2), NULL);
-    FieldTask_InitCall(taskMan, sub_02050CA8, v1);
+    encounter = NewEncounter(dto, EncEffects_CutInEffect(dto), EncEffects_BGM(dto), NULL);
+    FieldTask_InitCall(task, FieldTask_LinkEncounter, encounter);
 }
 
-static int sub_020516C8(const BattleRegulation *param0, int param1)
+static int sub_020516C8(const BattleRegulation *regulation, int battleType)
 {
-    int v0, v1;
+    int v0;
+    int v1 = sub_020261B0(regulation);
 
-    v1 = sub_020261B0(param0);
-
-    if (param1 & 0x8) {
+    if (battleType & BATTLE_TYPE_2vs2) {
         v0 = (UnkEnum_0202F510_14);
-    } else if (param1 & 0x2) {
+    } else if (battleType & BATTLE_TYPE_DOUBLES) {
         v0 = (UnkEnum_0202F510_07);
     } else {
         v0 = (UnkEnum_0202F510_00);
     }
 
-    if (v1 != 0xff) {
+    if (v1 != 0xFF) {
         v0 += 1 + v1;
     }
 
     return v0;
 }
 
-void sub_020516F4(FieldTask *taskMan, int param1, int param2, int param3)
+void Encounter_NewVsWiFi(FieldTask *task, int param1, int normalizedLevel, int wifiBattleType)
 {
-    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(taskMan);
-    UnkStruct_02050ACC *v1;
-    FieldBattleDTO *v2;
-    int v3;
-    int v4, v5;
-
-    if (param3 == 0) {
-        v4 = (0x4 | 0x1);
-        v2 = FieldBattleDTO_New(11, (0x4 | 0x1));
+    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(task);
+    Encounter *encounter;
+    FieldBattleDTO *dto;
+    int battleType, recordingResultCode, v5;
+
+    if (wifiBattleType == 0) {
```

Comment:
**question:** Is this something we have enough information about to make descriptive constants at this time?

Reply:
Not really, in my opinion, which is unfortunate. But, this function is only invoked in one place, so it's relatively easy to track down whenever that calling function is documented.